// api-credito-pessoal-demo/src/model/Body.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.Body = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The Body model module.
   * @module model/Body
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>Body</code>.
   * @alias module:model/Body
   * @class
   */
  var exports = function() {
    var _this = this;






  };

  /**
   * Constructs a <code>Body</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Body} obj Optional instance to populate.
   * @return {module:model/Body} The populated <code>Body</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('data-primeira-parcela')) {
        obj['data-primeira-parcela'] = ApiClient.convertToType(data['data-primeira-parcela'], 'String');
      }
      if (data.hasOwnProperty('seguro')) {
        obj['seguro'] = ApiClient.convertToType(data['seguro'], 'String');
      }
      if (data.hasOwnProperty('produto')) {
        obj['produto'] = ApiClient.convertToType(data['produto'], 'String');
      }
      if (data.hasOwnProperty('valor-desejado')) {
        obj['valor-desejado'] = ApiClient.convertToType(data['valor-desejado'], 'Number');
      }
      if (data.hasOwnProperty('prazo')) {
        obj['prazo'] = ApiClient.convertToType(data['prazo'], 'Number');
      }
    }
    return obj;
  }

  /**
   * @member {String} data-primeira-parcela
   */
  exports.prototype['data-primeira-parcela'] = undefined;
  /**
   * @member {module:model/Body.SeguroEnum} seguro
   */
  exports.prototype['seguro'] = undefined;
  /**
   * @member {String} produto
   */
  exports.prototype['produto'] = undefined;
  /**
   * @member {Number} valor-desejado
   */
  exports.prototype['valor-desejado'] = undefined;
  /**
   * @member {Number} prazo
   */
  exports.prototype['prazo'] = undefined;


  /**
   * Allowed values for the <code>seguro</code> property.
   * @enum {String}
   * @readonly
   */
  exports.SeguroEnum = {
    /**
     * value: "SIM"
     * @const
     */
    "SIM": "SIM",
    /**
     * value: "NAO"
     * @const
     */
    "NAO": "NAO"  };


  return exports;
}));
