// api-credito-pessoal-demo/src/model/CamposclienteSituacaoprofissionalEnderecocomercial.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposclienteSituacaoprofissionalEnderecocomercial = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposclienteSituacaoprofissionalEnderecocomercial model module.
   * @module model/CamposclienteSituacaoprofissionalEnderecocomercial
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposclienteSituacaoprofissionalEnderecocomercial</code>.
   * @alias module:model/CamposclienteSituacaoprofissionalEnderecocomercial
   * @class
   */
  var exports = function() {
    var _this = this;








  };

  /**
   * Constructs a <code>CamposclienteSituacaoprofissionalEnderecocomercial</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposclienteSituacaoprofissionalEnderecocomercial} obj Optional instance to populate.
   * @return {module:model/CamposclienteSituacaoprofissionalEnderecocomercial} The populated <code>CamposclienteSituacaoprofissionalEnderecocomercial</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('logradouro')) {
        obj['logradouro'] = ApiClient.convertToType(data['logradouro'], 'String');
      }
      if (data.hasOwnProperty('numero')) {
        obj['numero'] = ApiClient.convertToType(data['numero'], 'String');
      }
      if (data.hasOwnProperty('complemento')) {
        obj['complemento'] = ApiClient.convertToType(data['complemento'], 'String');
      }
      if (data.hasOwnProperty('bairro')) {
        obj['bairro'] = ApiClient.convertToType(data['bairro'], 'String');
      }
      if (data.hasOwnProperty('cidade')) {
        obj['cidade'] = ApiClient.convertToType(data['cidade'], 'String');
      }
      if (data.hasOwnProperty('uf')) {
        obj['uf'] = ApiClient.convertToType(data['uf'], 'String');
      }
      if (data.hasOwnProperty('cep')) {
        obj['cep'] = ApiClient.convertToType(data['cep'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} logradouro
   */
  exports.prototype['logradouro'] = undefined;
  /**
   * @member {String} numero
   */
  exports.prototype['numero'] = undefined;
  /**
   * @member {String} complemento
   */
  exports.prototype['complemento'] = undefined;
  /**
   * @member {String} bairro
   */
  exports.prototype['bairro'] = undefined;
  /**
   * @member {String} cidade
   */
  exports.prototype['cidade'] = undefined;
  /**
   * @member {String} uf
   */
  exports.prototype['uf'] = undefined;
  /**
   * @member {String} cep
   */
  exports.prototype['cep'] = undefined;



  return exports;
}));
