// api-credito-pessoal-demo/src/model/CamposclienteDocumentos.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposclienteDocumentos = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposclienteDocumentos model module.
   * @module model/CamposclienteDocumentos
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposclienteDocumentos</code>.
   * @alias module:model/CamposclienteDocumentos
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>CamposclienteDocumentos</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposclienteDocumentos} obj Optional instance to populate.
   * @return {module:model/CamposclienteDocumentos} The populated <code>CamposclienteDocumentos</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('numero')) {
        obj['numero'] = ApiClient.convertToType(data['numero'], 'String');
      }
      if (data.hasOwnProperty('tipo')) {
        obj['tipo'] = ApiClient.convertToType(data['tipo'], 'String');
      }
      if (data.hasOwnProperty('uf')) {
        obj['uf'] = ApiClient.convertToType(data['uf'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} numero
   */
  exports.prototype['numero'] = undefined;
  /**
   * @member {module:model/CamposclienteDocumentos.TipoEnum} tipo
   */
  exports.prototype['tipo'] = undefined;
  /**
   * @member {String} uf
   */
  exports.prototype['uf'] = undefined;


  /**
   * Allowed values for the <code>tipo</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TipoEnum = {
    /**
     * value: "RG"
     * @const
     */
    "RG": "RG",
    /**
     * value: "CNH"
     * @const
     */
    "CNH": "CNH",
    /**
     * value: "OAB"
     * @const
     */
    "OAB": "OAB"  };


  return exports;
}));
