// api-credito-pessoal-demo/src/model/CamposclienteTelefones.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposclienteTelefones = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposclienteTelefones model module.
   * @module model/CamposclienteTelefones
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposclienteTelefones</code>.
   * @alias module:model/CamposclienteTelefones
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>CamposclienteTelefones</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposclienteTelefones} obj Optional instance to populate.
   * @return {module:model/CamposclienteTelefones} The populated <code>CamposclienteTelefones</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('tipo')) {
        obj['tipo'] = ApiClient.convertToType(data['tipo'], 'String');
      }
      if (data.hasOwnProperty('ddd')) {
        obj['ddd'] = ApiClient.convertToType(data['ddd'], 'String');
      }
      if (data.hasOwnProperty('numero')) {
        obj['numero'] = ApiClient.convertToType(data['numero'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/CamposclienteTelefones.TipoEnum} tipo
   */
  exports.prototype['tipo'] = undefined;
  /**
   * @member {String} ddd
   */
  exports.prototype['ddd'] = undefined;
  /**
   * @member {String} numero
   */
  exports.prototype['numero'] = undefined;


  /**
   * Allowed values for the <code>tipo</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TipoEnum = {
    /**
     * value: "CELULAR"
     * @const
     */
    "CELULAR": "CELULAR",
    /**
     * value: "FIXO"
     * @const
     */
    "FIXO": "FIXO",
    /**
     * value: "COMERCIAL"
     * @const
     */
    "COMERCIAL": "COMERCIAL"  };


  return exports;
}));
