// api-credito-pessoal-demo/src/model/CamposclienteRedessociais.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposclienteRedessociais = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposclienteRedessociais model module.
   * @module model/CamposclienteRedessociais
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposclienteRedessociais</code>.
   * @alias module:model/CamposclienteRedessociais
   * @class
   */
  var exports = function() {
    var _this = this;



  };

  /**
   * Constructs a <code>CamposclienteRedessociais</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposclienteRedessociais} obj Optional instance to populate.
   * @return {module:model/CamposclienteRedessociais} The populated <code>CamposclienteRedessociais</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('nome')) {
        obj['nome'] = ApiClient.convertToType(data['nome'], 'String');
      }
      if (data.hasOwnProperty('conta')) {
        obj['conta'] = ApiClient.convertToType(data['conta'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/CamposclienteRedessociais.NomeEnum} nome
   */
  exports.prototype['nome'] = undefined;
  /**
   * @member {String} conta
   */
  exports.prototype['conta'] = undefined;


  /**
   * Allowed values for the <code>nome</code> property.
   * @enum {String}
   * @readonly
   */
  exports.NomeEnum = {
    /**
     * value: "FACEBOOK"
     * @const
     */
    "FACEBOOK": "FACEBOOK",
    /**
     * value: "TWITTER"
     * @const
     */
    "TWITTER": "TWITTER",
    /**
     * value: "INSTAGRAM"
     * @const
     */
    "INSTAGRAM": "INSTAGRAM",
    /**
     * value: "LINKEDIN"
     * @const
     */
    "LINKEDIN": "LINKEDIN"  };


  return exports;
}));
