// api-credito-pessoal-demo/src/model/CamposCliente.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/CamposclienteContasbancarias', 'model/CamposclienteDocumentos', 'model/CamposclienteEnderecoresidencial', 'model/CamposclienteRedessociais', 'model/CamposclienteSituacaoprofissional', 'model/CamposclienteTelefones'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./CamposclienteContasbancarias'), require('./CamposclienteDocumentos'), require('./CamposclienteEnderecoresidencial'), require('./CamposclienteRedessociais'), require('./CamposclienteSituacaoprofissional'), require('./CamposclienteTelefones'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposCliente = factory(root.ApiCrditoPessoalDemo.ApiClient, root.ApiCrditoPessoalDemo.CamposclienteContasbancarias, root.ApiCrditoPessoalDemo.CamposclienteDocumentos, root.ApiCrditoPessoalDemo.CamposclienteEnderecoresidencial, root.ApiCrditoPessoalDemo.CamposclienteRedessociais, root.ApiCrditoPessoalDemo.CamposclienteSituacaoprofissional, root.ApiCrditoPessoalDemo.CamposclienteTelefones);
  }
}(this, function(ApiClient, CamposclienteContasbancarias, CamposclienteDocumentos, CamposclienteEnderecoresidencial, CamposclienteRedessociais, CamposclienteSituacaoprofissional, CamposclienteTelefones) {
  'use strict';




  /**
   * The CamposCliente model module.
   * @module model/CamposCliente
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposCliente</code>.
   * @alias module:model/CamposCliente
   * @class
   */
  var exports = function() {
    var _this = this;















  };

  /**
   * Constructs a <code>CamposCliente</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposCliente} obj Optional instance to populate.
   * @return {module:model/CamposCliente} The populated <code>CamposCliente</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('nome')) {
        obj['nome'] = ApiClient.convertToType(data['nome'], 'String');
      }
      if (data.hasOwnProperty('cpf')) {
        obj['cpf'] = ApiClient.convertToType(data['cpf'], 'String');
      }
      if (data.hasOwnProperty('data-nascimento')) {
        obj['data-nascimento'] = ApiClient.convertToType(data['data-nascimento'], 'String');
      }
      if (data.hasOwnProperty('email')) {
        obj['email'] = ApiClient.convertToType(data['email'], 'String');
      }
      if (data.hasOwnProperty('redes-sociais')) {
        obj['redes-sociais'] = ApiClient.convertToType(data['redes-sociais'], [CamposclienteRedessociais]);
      }
      if (data.hasOwnProperty('escolaridade')) {
        obj['escolaridade'] = ApiClient.convertToType(data['escolaridade'], 'String');
      }
      if (data.hasOwnProperty('documentos')) {
        obj['documentos'] = ApiClient.convertToType(data['documentos'], [CamposclienteDocumentos]);
      }
      if (data.hasOwnProperty('estado-civil')) {
        obj['estado-civil'] = ApiClient.convertToType(data['estado-civil'], 'String');
      }
      if (data.hasOwnProperty('sexo')) {
        obj['sexo'] = ApiClient.convertToType(data['sexo'], 'String');
      }
      if (data.hasOwnProperty('endereco-residencial')) {
        obj['endereco-residencial'] = CamposclienteEnderecoresidencial.constructFromObject(data['endereco-residencial']);
      }
      if (data.hasOwnProperty('negativado')) {
        obj['negativado'] = ApiClient.convertToType(data['negativado'], 'String');
      }
      if (data.hasOwnProperty('contas-bancarias')) {
        obj['contas-bancarias'] = ApiClient.convertToType(data['contas-bancarias'], [CamposclienteContasbancarias]);
      }
      if (data.hasOwnProperty('telefones')) {
        obj['telefones'] = ApiClient.convertToType(data['telefones'], [CamposclienteTelefones]);
      }
      if (data.hasOwnProperty('situacao-profissional')) {
        obj['situacao-profissional'] = CamposclienteSituacaoprofissional.constructFromObject(data['situacao-profissional']);
      }
    }
    return obj;
  }

  /**
   * @member {String} nome
   */
  exports.prototype['nome'] = undefined;
  /**
   * @member {String} cpf
   */
  exports.prototype['cpf'] = undefined;
  /**
   * @member {String} data-nascimento
   */
  exports.prototype['data-nascimento'] = undefined;
  /**
   * @member {String} email
   */
  exports.prototype['email'] = undefined;
  /**
   * @member {Array.<module:model/CamposclienteRedessociais>} redes-sociais
   */
  exports.prototype['redes-sociais'] = undefined;
  /**
   * @member {module:model/CamposCliente.EscolaridadeEnum} escolaridade
   */
  exports.prototype['escolaridade'] = undefined;
  /**
   * @member {Array.<module:model/CamposclienteDocumentos>} documentos
   */
  exports.prototype['documentos'] = undefined;
  /**
   * @member {module:model/CamposCliente.EstadoCivilEnum} estado-civil
   */
  exports.prototype['estado-civil'] = undefined;
  /**
   * @member {module:model/CamposCliente.SexoEnum} sexo
   */
  exports.prototype['sexo'] = undefined;
  /**
   * @member {module:model/CamposclienteEnderecoresidencial} endereco-residencial
   */
  exports.prototype['endereco-residencial'] = undefined;
  /**
   * @member {module:model/CamposCliente.NegativadoEnum} negativado
   */
  exports.prototype['negativado'] = undefined;
  /**
   * @member {Array.<module:model/CamposclienteContasbancarias>} contas-bancarias
   */
  exports.prototype['contas-bancarias'] = undefined;
  /**
   * @member {Array.<module:model/CamposclienteTelefones>} telefones
   */
  exports.prototype['telefones'] = undefined;
  /**
   * @member {module:model/CamposclienteSituacaoprofissional} situacao-profissional
   */
  exports.prototype['situacao-profissional'] = undefined;


  /**
   * Allowed values for the <code>escolaridade</code> property.
   * @enum {String}
   * @readonly
   */
  exports.EscolaridadeEnum = {
    /**
     * value: "ANALFABETO"
     * @const
     */
    "ANALFABETO": "ANALFABETO",
    /**
     * value: "PRIMARIO"
     * @const
     */
    "PRIMARIO": "PRIMARIO",
    /**
     * value: "MEDIO"
     * @const
     */
    "MEDIO": "MEDIO",
    /**
     * value: "SUPERIOR"
     * @const
     */
    "SUPERIOR": "SUPERIOR",
    /**
     * value: "POS"
     * @const
     */
    "POS": "POS"  };

  /**
   * Allowed values for the <code>estado-civil</code> property.
   * @enum {String}
   * @readonly
   */
  exports.EstadoCivilEnum = {
    /**
     * value: "SOLTEIRO"
     * @const
     */
    "SOLTEIRO": "SOLTEIRO",
    /**
     * value: "CASADO"
     * @const
     */
    "CASADO": "CASADO",
    /**
     * value: "DIVORCIADO"
     * @const
     */
    "DIVORCIADO": "DIVORCIADO",
    /**
     * value: "VIUVO"
     * @const
     */
    "VIUVO": "VIUVO"  };

  /**
   * Allowed values for the <code>sexo</code> property.
   * @enum {String}
   * @readonly
   */
  exports.SexoEnum = {
    /**
     * value: "MASCULINO"
     * @const
     */
    "MASCULINO": "MASCULINO",
    /**
     * value: "FEMININO"
     * @const
     */
    "FEMININO": "FEMININO",
    /**
     * value: "OUTROS"
     * @const
     */
    "OUTROS": "OUTROS"  };

  /**
   * Allowed values for the <code>negativado</code> property.
   * @enum {String}
   * @readonly
   */
  exports.NegativadoEnum = {
    /**
     * value: "SIM"
     * @const
     */
    "SIM": "SIM",
    /**
     * value: "NAO"
     * @const
     */
    "NAO": "NAO"  };


  return exports;
}));
