// api-credito-pessoal-demo/src/model/CamposclienteContasbancarias.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposclienteContasbancarias = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposclienteContasbancarias model module.
   * @module model/CamposclienteContasbancarias
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposclienteContasbancarias</code>.
   * @alias module:model/CamposclienteContasbancarias
   * @class
   */
  var exports = function() {
    var _this = this;





  };

  /**
   * Constructs a <code>CamposclienteContasbancarias</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposclienteContasbancarias} obj Optional instance to populate.
   * @return {module:model/CamposclienteContasbancarias} The populated <code>CamposclienteContasbancarias</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('codigo-banco')) {
        obj['codigo-banco'] = ApiClient.convertToType(data['codigo-banco'], 'String');
      }
      if (data.hasOwnProperty('tipo-conta')) {
        obj['tipo-conta'] = ApiClient.convertToType(data['tipo-conta'], 'String');
      }
      if (data.hasOwnProperty('agencia')) {
        obj['agencia'] = ApiClient.convertToType(data['agencia'], 'String');
      }
      if (data.hasOwnProperty('conta')) {
        obj['conta'] = ApiClient.convertToType(data['conta'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} codigo-banco
   */
  exports.prototype['codigo-banco'] = undefined;
  /**
   * @member {module:model/CamposclienteContasbancarias.TipoContaEnum} tipo-conta
   */
  exports.prototype['tipo-conta'] = undefined;
  /**
   * @member {String} agencia
   */
  exports.prototype['agencia'] = undefined;
  /**
   * @member {String} conta
   */
  exports.prototype['conta'] = undefined;


  /**
   * Allowed values for the <code>tipo-conta</code> property.
   * @enum {String}
   * @readonly
   */
  exports.TipoContaEnum = {
    /**
     * value: "CONTA_CORRENTE"
     * @const
     */
    "CONTA_CORRENTE": "CONTA_CORRENTE",
    /**
     * value: "POUPANCA"
     * @const
     */
    "POUPANCA": "POUPANCA"  };


  return exports;
}));
