// api-credito-pessoal-demo/src/model/CamposcetCet.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.CamposcetCet = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CamposcetCet model module.
   * @module model/CamposcetCet
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>CamposcetCet</code>.
   * @alias module:model/CamposcetCet
   * @class
   */
  var exports = function() {
    var _this = this;

















  };

  /**
   * Constructs a <code>CamposcetCet</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CamposcetCet} obj Optional instance to populate.
   * @return {module:model/CamposcetCet} The populated <code>CamposcetCet</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('cet-anual')) {
        obj['cet-anual'] = ApiClient.convertToType(data['cet-anual'], 'Number');
      }
      if (data.hasOwnProperty('cet-mensal')) {
        obj['cet-mensal'] = ApiClient.convertToType(data['cet-mensal'], 'Number');
      }
      if (data.hasOwnProperty('produto')) {
        obj['produto'] = ApiClient.convertToType(data['produto'], 'String');
      }
      if (data.hasOwnProperty('prazo')) {
        obj['prazo'] = ApiClient.convertToType(data['prazo'], 'Number');
      }
      if (data.hasOwnProperty('data-primeira-parcela')) {
        obj['data-primeira-parcela'] = ApiClient.convertToType(data['data-primeira-parcela'], 'String');
      }
      if (data.hasOwnProperty('data-ultima-parcela')) {
        obj['data-ultima-parcela'] = ApiClient.convertToType(data['data-ultima-parcela'], 'String');
      }
      if (data.hasOwnProperty('iof')) {
        obj['iof'] = ApiClient.convertToType(data['iof'], 'Number');
      }
      if (data.hasOwnProperty('taxa')) {
        obj['taxa'] = ApiClient.convertToType(data['taxa'], 'Number');
      }
      if (data.hasOwnProperty('taxa-anual')) {
        obj['taxa-anual'] = ApiClient.convertToType(data['taxa-anual'], 'Number');
      }
      if (data.hasOwnProperty('valor-compra')) {
        obj['valor-compra'] = ApiClient.convertToType(data['valor-compra'], 'Number');
      }
      if (data.hasOwnProperty('valor-tc')) {
        obj['valor-tc'] = ApiClient.convertToType(data['valor-tc'], 'Number');
      }
      if (data.hasOwnProperty('valor-financiado')) {
        obj['valor-financiado'] = ApiClient.convertToType(data['valor-financiado'], 'Number');
      }
      if (data.hasOwnProperty('valor-liberado')) {
        obj['valor-liberado'] = ApiClient.convertToType(data['valor-liberado'], 'Number');
      }
      if (data.hasOwnProperty('valor-parcela')) {
        obj['valor-parcela'] = ApiClient.convertToType(data['valor-parcela'], 'Number');
      }
      if (data.hasOwnProperty('valor-seguro')) {
        obj['valor-seguro'] = ApiClient.convertToType(data['valor-seguro'], 'Number');
      }
      if (data.hasOwnProperty('valor-total')) {
        obj['valor-total'] = ApiClient.convertToType(data['valor-total'], 'Number');
      }
    }
    return obj;
  }

  /**
   * @member {Number} cet-anual
   */
  exports.prototype['cet-anual'] = undefined;
  /**
   * @member {Number} cet-mensal
   */
  exports.prototype['cet-mensal'] = undefined;
  /**
   * @member {String} produto
   */
  exports.prototype['produto'] = undefined;
  /**
   * @member {Number} prazo
   */
  exports.prototype['prazo'] = undefined;
  /**
   * @member {String} data-primeira-parcela
   */
  exports.prototype['data-primeira-parcela'] = undefined;
  /**
   * @member {String} data-ultima-parcela
   */
  exports.prototype['data-ultima-parcela'] = undefined;
  /**
   * @member {Number} iof
   */
  exports.prototype['iof'] = undefined;
  /**
   * @member {Number} taxa
   */
  exports.prototype['taxa'] = undefined;
  /**
   * @member {Number} taxa-anual
   */
  exports.prototype['taxa-anual'] = undefined;
  /**
   * @member {Number} valor-compra
   */
  exports.prototype['valor-compra'] = undefined;
  /**
   * @member {Number} valor-tc
   */
  exports.prototype['valor-tc'] = undefined;
  /**
   * @member {Number} valor-financiado
   */
  exports.prototype['valor-financiado'] = undefined;
  /**
   * @member {Number} valor-liberado
   */
  exports.prototype['valor-liberado'] = undefined;
  /**
   * @member {Number} valor-parcela
   */
  exports.prototype['valor-parcela'] = undefined;
  /**
   * @member {Number} valor-seguro
   */
  exports.prototype['valor-seguro'] = undefined;
  /**
   * @member {Number} valor-total
   */
  exports.prototype['valor-total'] = undefined;



  return exports;
}));
