// api-credito-pessoal-demo/src/model/InlineResponse2003Pendencias.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.InlineResponse2003Pendencias = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The InlineResponse2003Pendencias model module.
   * @module model/InlineResponse2003Pendencias
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>InlineResponse2003Pendencias</code>.
   * @alias module:model/InlineResponse2003Pendencias
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>InlineResponse2003Pendencias</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/InlineResponse2003Pendencias} obj Optional instance to populate.
   * @return {module:model/InlineResponse2003Pendencias} The populated <code>InlineResponse2003Pendencias</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('data')) {
        obj['data'] = ApiClient.convertToType(data['data'], 'String');
      }
      if (data.hasOwnProperty('parecer')) {
        obj['parecer'] = ApiClient.convertToType(data['parecer'], 'String');
      }
      if (data.hasOwnProperty('descricao')) {
        obj['descricao'] = ApiClient.convertToType(data['descricao'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} data
   */
  exports.prototype['data'] = undefined;
  /**
   * @member {String} parecer
   */
  exports.prototype['parecer'] = undefined;
  /**
   * @member {String} descricao
   */
  exports.prototype['descricao'] = undefined;



  return exports;
}));
