// api-credito-pessoal-demo/src/model/InlineResponse20011Parcelas.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.InlineResponse20011Parcelas = factory(root.ApiCrditoPessoalDemo.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The InlineResponse20011Parcelas model module.
   * @module model/InlineResponse20011Parcelas
   * @version 1.0.0
   */

  /**
   * Constructs a new <code>InlineResponse20011Parcelas</code>.
   * @alias module:model/InlineResponse20011Parcelas
   * @class
   */
  var exports = function() {
    var _this = this;










  };

  /**
   * Constructs a <code>InlineResponse20011Parcelas</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/InlineResponse20011Parcelas} obj Optional instance to populate.
   * @return {module:model/InlineResponse20011Parcelas} The populated <code>InlineResponse20011Parcelas</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('parcela')) {
        obj['parcela'] = ApiClient.convertToType(data['parcela'], 'String');
      }
      if (data.hasOwnProperty('iof')) {
        obj['iof'] = ApiClient.convertToType(data['iof'], 'Number');
      }
      if (data.hasOwnProperty('iff')) {
        obj['iff'] = ApiClient.convertToType(data['iff'], 'Number');
      }
      if (data.hasOwnProperty('pmt')) {
        obj['pmt'] = ApiClient.convertToType(data['pmt'], 'Number');
      }
      if (data.hasOwnProperty('principal')) {
        obj['principal'] = ApiClient.convertToType(data['principal'], 'Number');
      }
      if (data.hasOwnProperty('renda')) {
        obj['renda'] = ApiClient.convertToType(data['renda'], 'Number');
      }
      if (data.hasOwnProperty('saldo')) {
        obj['saldo'] = ApiClient.convertToType(data['saldo'], 'Number');
      }
      if (data.hasOwnProperty('situacao_parcela')) {
        obj['situacao_parcela'] = ApiClient.convertToType(data['situacao_parcela'], 'String');
      }
      if (data.hasOwnProperty('vcto')) {
        obj['vcto'] = ApiClient.convertToType(data['vcto'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} parcela
   */
  exports.prototype['parcela'] = undefined;
  /**
   * @member {Number} iof
   */
  exports.prototype['iof'] = undefined;
  /**
   * @member {Number} iff
   */
  exports.prototype['iff'] = undefined;
  /**
   * @member {Number} pmt
   */
  exports.prototype['pmt'] = undefined;
  /**
   * @member {Number} principal
   */
  exports.prototype['principal'] = undefined;
  /**
   * @member {Number} renda
   */
  exports.prototype['renda'] = undefined;
  /**
   * @member {Number} saldo
   */
  exports.prototype['saldo'] = undefined;
  /**
   * @member {String} situacao_parcela
   */
  exports.prototype['situacao_parcela'] = undefined;
  /**
   * @member {String} vcto
   */
  exports.prototype['vcto'] = undefined;



  return exports;
}));
