// api-credito-pessoal-demo/src/api/ClientesApi.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/Error', 'model/ErrorValidation', 'model/InlineResponse20010', 'model/InlineResponse20011'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/Error'), require('../model/ErrorValidation'), require('../model/InlineResponse20010'), require('../model/InlineResponse20011'));
  } else {
    // Browser globals (root is window)
    if (!root.ApiCrditoPessoalDemo) {
      root.ApiCrditoPessoalDemo = {};
    }
    root.ApiCrditoPessoalDemo.ClientesApi = factory(root.ApiCrditoPessoalDemo.ApiClient, root.ApiCrditoPessoalDemo.Error, root.ApiCrditoPessoalDemo.ErrorValidation, root.ApiCrditoPessoalDemo.InlineResponse20010, root.ApiCrditoPessoalDemo.InlineResponse20011);
  }
}(this, function(ApiClient, Error, ErrorValidation, InlineResponse20010, InlineResponse20011) {
  'use strict';

  /**
   * Clientes service.
   * @module api/ClientesApi
   * @version 1.0.0
   */

  /**
   * Constructs a new ClientesApi. 
   * @alias module:api/ClientesApi
   * @class
   * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the clientesCpfContratosGet operation.
     * @callback module:api/ClientesApi~clientesCpfContratosGetCallback
     * @param {String} error Error message, if any.
     * @param {module:model/InlineResponse20011} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Consulta todos os contratos ativos de um cliente
     * @param {String} clientId 
     * @param {Number} cpf CPF de uma pessoa que se deseja consultar
     * @param {module:api/ClientesApi~clientesCpfContratosGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/InlineResponse20011}
     */
    this.clientesCpfContratosGet = function(clientId, cpf, callback) {
      var postBody = null;

      // verify the required parameter 'clientId' is set
      if (clientId === undefined || clientId === null) {
        throw new Error("Missing the required parameter 'clientId' when calling clientesCpfContratosGet");
      }

      // verify the required parameter 'cpf' is set
      if (cpf === undefined || cpf === null) {
        throw new Error("Missing the required parameter 'cpf' when calling clientesCpfContratosGet");
      }


      var pathParams = {
        'cpf': cpf
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
        'client_id': clientId
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = InlineResponse20011;

      return this.apiClient.callApi(
        '/clientes/{cpf}/contratos', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the clientesCpfGet operation.
     * @callback module:api/ClientesApi~clientesCpfGetCallback
     * @param {String} error Error message, if any.
     * @param {module:model/InlineResponse20010} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Consulta dados de um cliente
     * @param {String} clientId 
     * @param {Number} cpf CPF de uma pessoa que se deseja consultar
     * @param {module:api/ClientesApi~clientesCpfGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/InlineResponse20010}
     */
    this.clientesCpfGet = function(clientId, cpf, callback) {
      var postBody = null;

      // verify the required parameter 'clientId' is set
      if (clientId === undefined || clientId === null) {
        throw new Error("Missing the required parameter 'clientId' when calling clientesCpfGet");
      }

      // verify the required parameter 'cpf' is set
      if (cpf === undefined || cpf === null) {
        throw new Error("Missing the required parameter 'cpf' when calling clientesCpfGet");
      }


      var pathParams = {
        'cpf': cpf
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
        'client_id': clientId
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = InlineResponse20010;

      return this.apiClient.callApi(
        '/clientes/{cpf}', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
