// api-credito-pessoal-demo/src/index.js
/**
 * API Crédito Pessoal - Demo
 * API para tratamento de propostas de crédito pessoal utilizada para estudos do fluxo de crédito da Portocred
 *
 * OpenAPI spec version: 1.0.0
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.3.1
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/Body', 'model/Body1', 'model/Body2', 'model/CamposCet', 'model/CamposCliente', 'model/CamposProposta', 'model/CamposcetCet', 'model/CamposclienteContasbancarias', 'model/CamposclienteDocumentos', 'model/CamposclienteEnderecoresidencial', 'model/CamposclienteRedessociais', 'model/CamposclienteSituacaoprofissional', 'model/CamposclienteSituacaoprofissionalEnderecocomercial', 'model/CamposclienteTelefones', 'model/CampospropostaDadosadicionais', 'model/CampospropostaProposta', 'model/Error', 'model/ErrorValidation', 'model/InlineResponse200', 'model/InlineResponse2001', 'model/InlineResponse20010', 'model/InlineResponse20011', 'model/InlineResponse20011Metadata', 'model/InlineResponse20011Parcelas', 'model/InlineResponse20011Results', 'model/InlineResponse2002', 'model/InlineResponse2003', 'model/InlineResponse2003Pendencias', 'model/InlineResponse2004', 'model/InlineResponse2005', 'model/InlineResponse2006', 'model/InlineResponse2007', 'model/InlineResponse2008', 'model/InlineResponse2009', 'model/InlineResponse201', 'api/ClientesApi', 'api/ContratosApi', 'api/PropostasApi', 'api/SimulaesApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/Body'), require('./model/Body1'), require('./model/Body2'), require('./model/CamposCet'), require('./model/CamposCliente'), require('./model/CamposProposta'), require('./model/CamposcetCet'), require('./model/CamposclienteContasbancarias'), require('./model/CamposclienteDocumentos'), require('./model/CamposclienteEnderecoresidencial'), require('./model/CamposclienteRedessociais'), require('./model/CamposclienteSituacaoprofissional'), require('./model/CamposclienteSituacaoprofissionalEnderecocomercial'), require('./model/CamposclienteTelefones'), require('./model/CampospropostaDadosadicionais'), require('./model/CampospropostaProposta'), require('./model/Error'), require('./model/ErrorValidation'), require('./model/InlineResponse200'), require('./model/InlineResponse2001'), require('./model/InlineResponse20010'), require('./model/InlineResponse20011'), require('./model/InlineResponse20011Metadata'), require('./model/InlineResponse20011Parcelas'), require('./model/InlineResponse20011Results'), require('./model/InlineResponse2002'), require('./model/InlineResponse2003'), require('./model/InlineResponse2003Pendencias'), require('./model/InlineResponse2004'), require('./model/InlineResponse2005'), require('./model/InlineResponse2006'), require('./model/InlineResponse2007'), require('./model/InlineResponse2008'), require('./model/InlineResponse2009'), require('./model/InlineResponse201'), require('./api/ClientesApi'), require('./api/ContratosApi'), require('./api/PropostasApi'), require('./api/SimulaesApi'));
  }
}(function(ApiClient, Body, Body1, Body2, CamposCet, CamposCliente, CamposProposta, CamposcetCet, CamposclienteContasbancarias, CamposclienteDocumentos, CamposclienteEnderecoresidencial, CamposclienteRedessociais, CamposclienteSituacaoprofissional, CamposclienteSituacaoprofissionalEnderecocomercial, CamposclienteTelefones, CampospropostaDadosadicionais, CampospropostaProposta, Error, ErrorValidation, InlineResponse200, InlineResponse2001, InlineResponse20010, InlineResponse20011, InlineResponse20011Metadata, InlineResponse20011Parcelas, InlineResponse20011Results, InlineResponse2002, InlineResponse2003, InlineResponse2003Pendencias, InlineResponse2004, InlineResponse2005, InlineResponse2006, InlineResponse2007, InlineResponse2008, InlineResponse2009, InlineResponse201, ClientesApi, ContratosApi, PropostasApi, SimulaesApi) {
  'use strict';

  /**
   * API_para_tratamento_de_propostas_de_crdito_pessoal_utilizada_para_estudos_do_fluxo_de_crdito_da_Portocred.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var ApiCrditoPessoalDemo = require('index'); // See note below*.
   * var xxxSvc = new ApiCrditoPessoalDemo.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new ApiCrditoPessoalDemo.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new ApiCrditoPessoalDemo.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new ApiCrditoPessoalDemo.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 1.0.0
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The Body model constructor.
     * @property {module:model/Body}
     */
    Body: Body,
    /**
     * The Body1 model constructor.
     * @property {module:model/Body1}
     */
    Body1: Body1,
    /**
     * The Body2 model constructor.
     * @property {module:model/Body2}
     */
    Body2: Body2,
    /**
     * The CamposCet model constructor.
     * @property {module:model/CamposCet}
     */
    CamposCet: CamposCet,
    /**
     * The CamposCliente model constructor.
     * @property {module:model/CamposCliente}
     */
    CamposCliente: CamposCliente,
    /**
     * The CamposProposta model constructor.
     * @property {module:model/CamposProposta}
     */
    CamposProposta: CamposProposta,
    /**
     * The CamposcetCet model constructor.
     * @property {module:model/CamposcetCet}
     */
    CamposcetCet: CamposcetCet,
    /**
     * The CamposclienteContasbancarias model constructor.
     * @property {module:model/CamposclienteContasbancarias}
     */
    CamposclienteContasbancarias: CamposclienteContasbancarias,
    /**
     * The CamposclienteDocumentos model constructor.
     * @property {module:model/CamposclienteDocumentos}
     */
    CamposclienteDocumentos: CamposclienteDocumentos,
    /**
     * The CamposclienteEnderecoresidencial model constructor.
     * @property {module:model/CamposclienteEnderecoresidencial}
     */
    CamposclienteEnderecoresidencial: CamposclienteEnderecoresidencial,
    /**
     * The CamposclienteRedessociais model constructor.
     * @property {module:model/CamposclienteRedessociais}
     */
    CamposclienteRedessociais: CamposclienteRedessociais,
    /**
     * The CamposclienteSituacaoprofissional model constructor.
     * @property {module:model/CamposclienteSituacaoprofissional}
     */
    CamposclienteSituacaoprofissional: CamposclienteSituacaoprofissional,
    /**
     * The CamposclienteSituacaoprofissionalEnderecocomercial model constructor.
     * @property {module:model/CamposclienteSituacaoprofissionalEnderecocomercial}
     */
    CamposclienteSituacaoprofissionalEnderecocomercial: CamposclienteSituacaoprofissionalEnderecocomercial,
    /**
     * The CamposclienteTelefones model constructor.
     * @property {module:model/CamposclienteTelefones}
     */
    CamposclienteTelefones: CamposclienteTelefones,
    /**
     * The CampospropostaDadosadicionais model constructor.
     * @property {module:model/CampospropostaDadosadicionais}
     */
    CampospropostaDadosadicionais: CampospropostaDadosadicionais,
    /**
     * The CampospropostaProposta model constructor.
     * @property {module:model/CampospropostaProposta}
     */
    CampospropostaProposta: CampospropostaProposta,
    /**
     * The Error model constructor.
     * @property {module:model/Error}
     */
    Error: Error,
    /**
     * The ErrorValidation model constructor.
     * @property {module:model/ErrorValidation}
     */
    ErrorValidation: ErrorValidation,
    /**
     * The InlineResponse200 model constructor.
     * @property {module:model/InlineResponse200}
     */
    InlineResponse200: InlineResponse200,
    /**
     * The InlineResponse2001 model constructor.
     * @property {module:model/InlineResponse2001}
     */
    InlineResponse2001: InlineResponse2001,
    /**
     * The InlineResponse20010 model constructor.
     * @property {module:model/InlineResponse20010}
     */
    InlineResponse20010: InlineResponse20010,
    /**
     * The InlineResponse20011 model constructor.
     * @property {module:model/InlineResponse20011}
     */
    InlineResponse20011: InlineResponse20011,
    /**
     * The InlineResponse20011Metadata model constructor.
     * @property {module:model/InlineResponse20011Metadata}
     */
    InlineResponse20011Metadata: InlineResponse20011Metadata,
    /**
     * The InlineResponse20011Parcelas model constructor.
     * @property {module:model/InlineResponse20011Parcelas}
     */
    InlineResponse20011Parcelas: InlineResponse20011Parcelas,
    /**
     * The InlineResponse20011Results model constructor.
     * @property {module:model/InlineResponse20011Results}
     */
    InlineResponse20011Results: InlineResponse20011Results,
    /**
     * The InlineResponse2002 model constructor.
     * @property {module:model/InlineResponse2002}
     */
    InlineResponse2002: InlineResponse2002,
    /**
     * The InlineResponse2003 model constructor.
     * @property {module:model/InlineResponse2003}
     */
    InlineResponse2003: InlineResponse2003,
    /**
     * The InlineResponse2003Pendencias model constructor.
     * @property {module:model/InlineResponse2003Pendencias}
     */
    InlineResponse2003Pendencias: InlineResponse2003Pendencias,
    /**
     * The InlineResponse2004 model constructor.
     * @property {module:model/InlineResponse2004}
     */
    InlineResponse2004: InlineResponse2004,
    /**
     * The InlineResponse2005 model constructor.
     * @property {module:model/InlineResponse2005}
     */
    InlineResponse2005: InlineResponse2005,
    /**
     * The InlineResponse2006 model constructor.
     * @property {module:model/InlineResponse2006}
     */
    InlineResponse2006: InlineResponse2006,
    /**
     * The InlineResponse2007 model constructor.
     * @property {module:model/InlineResponse2007}
     */
    InlineResponse2007: InlineResponse2007,
    /**
     * The InlineResponse2008 model constructor.
     * @property {module:model/InlineResponse2008}
     */
    InlineResponse2008: InlineResponse2008,
    /**
     * The InlineResponse2009 model constructor.
     * @property {module:model/InlineResponse2009}
     */
    InlineResponse2009: InlineResponse2009,
    /**
     * The InlineResponse201 model constructor.
     * @property {module:model/InlineResponse201}
     */
    InlineResponse201: InlineResponse201,
    /**
     * The ClientesApi service constructor.
     * @property {module:api/ClientesApi}
     */
    ClientesApi: ClientesApi,
    /**
     * The ContratosApi service constructor.
     * @property {module:api/ContratosApi}
     */
    ContratosApi: ContratosApi,
    /**
     * The PropostasApi service constructor.
     * @property {module:api/PropostasApi}
     */
    PropostasApi: PropostasApi,
    /**
     * The SimulaesApi service constructor.
     * @property {module:api/SimulaesApi}
     */
    SimulaesApi: SimulaesApi
  };

  return exports;
}));
